backup/src/main/java/io/camunda/zeebe/backup/api/BackupStore.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.backup.api;
+
+import java.util.concurrent.CompletableFuture;
+
+/** A store where the backup is stored * */
+public interface BackupStore {
+
+  /**
+   * Saves the backup to the backup storage.
+   *
+   * @param backup
+   * @return future which will be completed after the backup is fully saved in the storage.
+   */
+  CompletableFuture<Void> save(Backup backup);
+
+  /**
+   * Returns the status of the backup
+   *
+   * @param metadata
+   * @return future with the status
+   */
+  CompletableFuture<BackupStatus> getStatus(BackupMetadata metadata);
+
+  /**
+   * Delete all state related to backup from the storage
+   *
+   * @param metadata
+   * @return future which will be completed after it is deleted
+   */
+  CompletableFuture<Void> delete(BackupMetadata metadata);
+
+  /**
+   * Restores the backup
+   *
+   * @param metadata
```

Comment:
:wrench: I'd rather leave out `@param` and `@return` as long as they don't document anything interesting.